### PR TITLE
Allow escaped commas in jmx.dump-tables config property

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxConnectorConfig.java [hunk 1]
```diff
@@ -41,7 +42,28 @@
     @Config("jmx.dump-tables")
     public JmxConnectorConfig setDumpTables(String tableNames)
     {
-        this.dumpTables = ImmutableSet.copyOf(Splitter.on(',').omitEmptyStrings().trimResults().split(tableNames));
+        ImmutableSet.Builder<String> dupTables = ImmutableSet.builder();
```
Comment (martint) [inline presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxConnectorConfig.java]: Typo: dupTables -> dumpTables

File: presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxConnectorConfig.java [hunk 2]
```diff
@@ -41,7 +42,28 @@
...
+        String tableName = "";
+        Iterable<String> parts = Splitter.on(',').omitEmptyStrings().split(tableNames);
```
Comment (martint) [inline presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxConnectorConfig.java]: How about:

```java
        this.dumpTables = Splitter.on(Pattern.compile("(?<!\\\\),")) // match "," not preceded by "\"
                .omitEmptyStrings()
                .splitToList(tableNames)
                .stream()
                .map(part -> part.replace("\\,", ",")) // unescape all escaped commas
                .collect(Collectors.toSet());
```

Basically, it splits on "," not preceded by "\", but it doesn't consider the preceding character as part of the split pattern (it uses regex negative look-behind).

Comment (losipiuk) [inline presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxConnectorConfig.java]: Ashamed :/ This looks much better. Updated to your version.

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**